web/components/swipe/swipe-card.tsx

```diff
@@ -0,0 +1,376 @@
+import clsx from 'clsx'
+
+import { track } from '@amplitude/analytics-browser'
+import { animated, useSpring } from '@react-spring/web'
+import { rubberbandIfOutOfBounds, useDrag } from '@use-gesture/react'
+import { getOutcomeProbabilityAfterBet } from 'common/calculate'
+import { BinaryContract, Contract } from 'common/contract'
+import { getBinaryProb } from 'common/contract-details'
+import { User } from 'common/user'
+import { memo, ReactNode, SetStateAction, useEffect, useState } from 'react'
+import { LikeButton } from 'web/components/contract/like-button'
+import { useContract } from 'web/hooks/use-contracts'
+import { placeBet } from 'web/lib/firebase/api'
+import { contractPath } from 'web/lib/firebase/contracts'
+import { logView } from 'web/lib/firebase/views'
+import { fromNow } from 'web/lib/util/time'
+import { Col } from '../layout/col'
+import { Avatar } from '../widgets/avatar'
+import { SiteLink } from '../widgets/site-link'
+import { SwipeBetPanel } from './swipe-bet-panel'
+import getQuestionSize, {
+  BUFFER_CARD_COLOR,
+  BUFFER_CARD_OPACITY,
+  isStatusAFailure,
+} from './swipe-helpers'
+import Percent, { DescriptionAndModal } from './swipe-widgets'
+
+const horizontalSwipeDist = 80
+const verticalSwipeDist = -100
+
+export type SwipeAction = 'left' | 'right' | 'up' | 'none'
+
+function getSwipeAction(
+  mx: number,
+  my: number,
+  cappedDist: number
+): SwipeAction {
+  if (cappedDist >= horizontalSwipeDist) {
+    return mx < 0 ? 'left' : 'right'
+  }
+  if (my <= verticalSwipeDist) {
+    return 'up'
```

Comment:
Why not support swiping back to the previous card?

Reply:
because we must always be looking forward

Reply:
lol I guess I could but I'll just have to think out logistic ux things, like will the card behind change depending on whether you swipe up or down (prev vs. next)